dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java

```diff
@@ -0,0 +1,511 @@
+    String prefix;
+    if (namespace.isEmpty()) {
+      prefix = warehouseLocation.name();
+    } else {
+      prefix = String.format("%s%s/", warehouseLocation.name(),
+          String.join("/", namespace.levels()));
+    }
+
+    return new EcsURI(warehouseLocation.bucket(), prefix);
+  }
+
+  private TableIdentifier parseTableId(Namespace namespace, EcsURI prefix, S3Object s3Object) {
+    String key = s3Object.getKey();
+    Preconditions.checkArgument(key.startsWith(prefix.name()),
+        "List result should have same prefix", key, prefix);
+
+    String tableName = key.substring(
+        prefix.name().length(),
+        key.length() - TABLE_OBJECT_SUFFIX.length());
+    return TableIdentifier.of(namespace, tableName);
+  }
+
+  /**
+   * Remove table object. If the purge flag is set, remove all data objects.
+   */
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    if (!tableExists(identifier)) {
+      throw new NoSuchTableException("Table %s does not exist", identifier);
+    }
+
+    EcsURI tableObjectURI = tableURI(identifier);
+    if (purge) {
+      // if re-use the same instance, current() will throw exception.
+      TableOperations ops = newTableOps(identifier);
+      TableMetadata current = ops.current();
+      if (current == null) {
+        return false;
+      }
+
+      CatalogUtil.dropTableData(ops.io(), current);
+    }
+
+    client.deleteObject(tableObjectURI.bucket(), tableObjectURI.name());
+    return true;
+  }
+
+  private EcsURI tableURI(TableIdentifier id) {
+    EcsURI prefix = namespacePrefix(id.namespace());
+    // The prefix has the delimiter at the tail.
+    return new EcsURI(prefix.bucket(), prefix.name() + id.name() + TABLE_OBJECT_SUFFIX);
+  }
+
+  /**
+   * Table rename will only move table object, the data objects will still be in-place.
+   *
+   * @param from identifier of the table to rename
+   * @param to   new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Cannot rename %s to %s because namespace %s does not exist",
+              from, to, to.namespace());
+    }
+
+    if (tableExists(to)) {
+      throw new AlreadyExistsException("Cannot rename %s because destination table %s exists", from, to);
+    }
+
+    EcsURI fromURI = tableURI(from);
+    if (!objectMetadata(fromURI).isPresent()) {
+      throw new NoSuchTableException("Cannot rename table because table %s does not exist", from);
+    }
+
+    Properties properties = loadProperties(fromURI);
+    EcsURI toURI = tableURI(to);
+
+    if (!putNewProperties(toURI, properties.content())) {
+      throw new AlreadyExistsException("Cannot rename %s because destination table %s exists", from, to);
+    }
+
+    client.deleteObject(fromURI.bucket(), fromURI.name());
+    LOG.info("rename table {} to {}", from, to);
```

Comment:
In the current approach, I'm concerning that we cannot guarantee the atomicity between the `putNewProperties` and `deleteObject`, right ?   (which is validating the iceberg transaction semantic )

Reply:
> In the current approach, I'm concerning that we cannot guarantee the atomicity between the `putNewProperties` and `deleteObject`, right ? (which is validating the iceberg transaction semantic )

Yes. It's not atomic.

Reply:
> Nit: It's recommended to uppercase the first character from the message `rename table {} to {}`.

Fixed.